From 0d3cbf6c68575c16ed42e0b183d1147ae9132433 Mon Sep 17 00:00:00 2001
From: Tyler Gunn <tgunn@google.com>
Date: Mon, 29 Oct 2018 09:35:49 -0700
Subject: [PATCH 20/39] Ensure ACTION_EMERGENCY_CALL_STATE_CHANGED doesn't send
 on GSM.

If on a CDMA carrier which supports ECM and roaming on GSM, we should not
be sending the ACTION_EMERGENCY_CALL_STATE_CHANGED intents since we will
not be going into ECM.

Test: Insert VZW SIM and dial emergency call, then check if WiFi can be turned off.
Bug: 117912348
Bug: 120583450

Change-Id: Ib2a9c5732fa9f81349ec93e8326d2e5753fed0f7
Merged-In: Ib2a9c5732fa9f81349ec93e8326d2e5753fed0f7
(cherry picked from commit 7f6a473ae03749e4915280d86744b38ade6bf331)
---
 src/java/com/android/internal/telephony/GsmCdmaPhone.java | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/java/com/android/internal/telephony/GsmCdmaPhone.java b/src/java/com/android/internal/telephony/GsmCdmaPhone.java
index 2a94fc5cf..db328a780 100644
--- a/src/java/com/android/internal/telephony/GsmCdmaPhone.java
+++ b/src/java/com/android/internal/telephony/GsmCdmaPhone.java
@@ -641,11 +641,16 @@ public class GsmCdmaPhone extends Phone {
         intent.putExtra(PhoneConstants.PHONE_IN_ECM_STATE, isInEcm());
         SubscriptionManager.putPhoneIdAndSubIdExtra(intent, getPhoneId());
         ActivityManager.broadcastStickyIntent(intent, UserHandle.USER_ALL);
-        if (DBG) logd("sendEmergencyCallbackModeChange");
+        logi("sendEmergencyCallbackModeChange");
     }
 
     @Override
     public void sendEmergencyCallStateChange(boolean callActive) {
+        if (!isPhoneTypeCdma()) {
+            // It possible that this method got called from ImsPhoneCallTracker#
+            logi("sendEmergencyCallbackModeChange - skip for non-cdma");
+            return;
+        }
         if (mBroadcastEmergencyCallStateChanges) {
             Intent intent = new Intent(TelephonyIntents.ACTION_EMERGENCY_CALL_STATE_CHANGED);
             intent.putExtra(PhoneConstants.PHONE_IN_EMERGENCY_CALL, callActive);
-- 
2.17.1

